### ISSUE TITLE
Links within docx comments not handled correctly

### ISSUE THREAD

**Issue by cathornburg** [jgm/pandoc#6743]
pandoc 2.10.1
Microsoft Word for Microsoft Office 365 ProPlus (however, I believe I also confirmed the issue with Office 2013)

Neither docx reader nor writer are handling hyperlinks within comments correctly.

## Docx Reader

To reproduce:
1. Create a docx document with a comment that includes a hyperlink within the comment content. The example [comment-test.docx](https://github.com/jgm/pandoc/files/5372192/comment-test.docx) uses [Pandoc](https://pandoc.org).
2. Use Pandoc to convert to any format that supports comments. For example:
`>pandoc --track-changes=all comment-test.docx -t native`
3. Observe that the resulting link is sometimes to the wrong URL:

**Actual output:**

[Para [Span ("",["comment-start"],[("id","0"),("author","Thornburg, Christopher A"),("date","2020-10-13T07:38:00Z")]) [Link ("",[],[]) [Str "Pandoc"] ("**styles.xml**",""),Space,Str "is",Space,Str "awesome!"],Str "Content",Space,Str "commented",Space,Str "with",Space,Str "a",Space,Str "link.",Span ("",["comment-end"],[("id","0")]) []]]

**Expected output:**

[Para [Span ("",["comment-start"],[("id","0"),("author","Thornburg, Christopher A"),("date","2020-10-13T07:38:00Z")]) [Link ("",[],[]) [Str "Pandoc"] ("**https://pandoc.org**",""),Space,Str "is",Space,Str "awesome!"],Str "Content",Space,Str "commented",Space,Str "with",Space,Str "a",Space,Str "link.",Span ("",["comment-end"],[("id","0")]) []]]

## Docx Writer

To reproduce:
1. Create a file with a comment that includes a hyperlink within the comment content. For example, [from-pandoc.txt](https://github.com/jgm/pandoc/files/5372288/from-pandoc.txt)
 is a Pandoc native format file containing a link to [Pandoc](https://pandoc.org) within a comment.
2. Use Pandoc to convert the file to docx. For example:
`>pandoc --track-changes=all from-pandoc.txt -f native -o to_docx.docx`
3. Open the resulting file ([to_docx.docx](https://github.com/jgm/pandoc/files/5372262/to_docx.docx)) with MS Word.
4. MS Word will report an error with the file. If you accept recovery, the file will open, the linked text within the comment will be formatted as a hyperlink, but the link itself contains no URL.

**Comment by jgm** [comment]
I'm not sure why this happens, but as a stopgap we might just strip out links from comments in the writer to avoid creating a bad docx.

**Comment by cathornburg** [comment]
It's even easy to create a filter to replace the links with text, but unfortunately, for our workflow, links within comments are critical, and I haven't been able to find a workaround to fix them without running a VB script within Word after the conversion.

**Comment by cathornburg** [comment]
I've found the bug, at least in the Docx Reader. The hyperlink handler always looks up hyperlink `rId`s in document relationships, but hyperlinks within comments actually reference comments relationships (specified in `_rels/comments.xml.rels`).

**Comment by jgm** [comment]
Yes, I figured it was something like that.  (docx format is so much fun!)
The writer probably has a parallel issue.

**Comment by cathornburg** [comment]
I was able to make simple fixes for both the reader and writer. I'll try to submit patches soon.

**Comment by tarleb** [comment]
@cathornburg if we can support you with this somehow, please do let us know. Maybe pushing your current progress would be a good step to get this over the finish line?
